tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/Languages/LanguageRepoService.cs

```diff
@@ -46,6 +46,12 @@ public interface ILanguageRepoService
     /// <param name="packagePath">Absolute path to the package directory</param>
     /// <returns>CLI check response containing success/failure status and response message</returns>
     Task<CLICheckResponse> ValidateChangelogAsync(string packagePath);
+
+    /// <summary>
+    /// A short identifier for the language this repo service handles (e.g. "java", "python").
+    /// This is used by the update service factory to select the correct UpdateLanguageService.
+    /// </summary>
+    string Language { get; }
```

Comment:
This feels like we're missing out on just taking advantage of polymorphism here. Instead of attempting to match outside of this class, just add the methods you need onto the base interface and update the individual LanguageRepo implementations.